database/database.go

```diff
@@ -61,6 +62,36 @@ type Config struct {
 	SSLMode        string `yaml:"ssl_mode"`
 }
 
+type BeforeQueryFunc func(ctx context.Context, query string, args ...interface{}) context.Context
+
+func (extDb *Database) SetBeforeQuery(f BeforeQueryFunc) {
```

Comment:
The name doesn't reflect what the function does. Either rename the function, or change its code in a way, when in sets the before func.